packages/office-ui-fabric-react/src/components/Panel/Panel.base.tsx

```diff
@@ -347,14 +348,20 @@ export class PanelBase extends BaseComponent<IPanelProps, IPanelState> implement
     defaultRender?: (props?: IPanelProps) => JSX.Element | null,
     headerTextId?: string | undefined
   ): JSX.Element | null => {
-    const { headerText } = props;
+    const { headerText, headerTextProps = {} } = props;
 
     if (headerText) {
       return (
         <div className={this._classNames.header}>
-          <p className={this._classNames.headerText} id={headerTextId} role="heading" aria-level={2}>
+          <div
+            role="heading"
+            aria-level={2}
+            {...headerTextProps}
+            className={css(this._classNames.headerText, headerTextProps.className)}
+            id={headerTextProps.id || headerTextId}
```

Comment:
If `headerTextProps.id` is given priority couldn't this line just move above 359 and be `id={headerTextId}`?